exodus_gw/models/publish.py

```diff
@@ -46,15 +46,19 @@ def resolve_links(self):
         ln_item_paths = [item.link_to for item in ln_items]
 
         # Store only necessary fields from matching items to conserve memory.
-        match = Bundle("match", Item.web_uri, Item.object_key)
+        match = Bundle(
+            "match", Item.web_uri, Item.object_key, Item.content_type
+        )
         matches = {
-            row.match["web_uri"]: row.match["object_key"]
+            row.match["web_uri"]: {
+                "object_key": row.match["object_key"],
+                "content_type": row.match["content_type"],
+            }
             for row in db.query(match).filter(Item.web_uri.in_(ln_item_paths))
         }
 
         for ln_item in ln_items:
-            ln_item.object_key = matches.get(ln_item.link_to)
-            if not ln_item.object_key:
+            if not matches:
```

Comment:
I think it still makes sense to look for the desired object_key here. Otherwise, this check will pass for any arbitrary object in `matches` with the potential to set ln_item values to NoneType. So, I think we'd want the code you've added below to be up here and then check if the object_key was actually set before raising, as was done before.
```suggestion
            match = matches.get(lin_item.link_to)
            ln_item.object_key = match.get("object_key")
            ln_item.content_type = match.get("content_type")
            if not ln_item.object_key or not ln_item.content_type:
```
Or at least find the correct match from `matches` and verify it has both an object_key and a content_type before passing the exception.